pkg/users/account-details.js

```diff
@@ -343,3 +334,113 @@
         </Page>
     );
 }
+
+export const AccountGroupsSelect = ({ name, groups, setError }) => {
+    const [isOpenGroup, setIsOpenGroup] = useState(false);
+    const [selected, setSelected] = useState();
+    const [validated, setValidated] = useState('default');
+    const [primaryGroupName, setPrimaryGroupName] = useState();
+    const previousValue = useRef(null);
+
+    useEffect(() => {
+        console.info({ groups });
+        const usedGroups = groups.filter(group => group.userlist.includes(name));
+        const primaryGroup = groups.find(group => group.userlistPrimary.includes(name));
+        const _primaryGroupName = primaryGroup ? primaryGroup.name : undefined;
+        const _selected = usedGroups.map(group => group.name);
+        if (primaryGroup)
+            _selected.push(_primaryGroupName);
+
+        const items_diff = previousValue.current
+            ? _selected
+                    .filter(x => !previousValue.current.includes(x))
+                    .concat(previousValue.current.filter(x => !_selected.includes(x)))
+            : [];
+        const validated = items_diff.length ? 'warning' : 'default';
+        previousValue.current = _selected;
+
+        setValidated(validated);
+        setSelected(_selected);
+        setPrimaryGroupName(_primaryGroupName);
+    }, [groups, setSelected, name, previousValue, setValidated]);
+
+    const removeGroup = group => {
+        return cockpit.spawn(["/usr/sbin/usermod", "-r", "-G", group, name], { superuser: "require", err: "message" })
+                .then(() => {
+                    setSelected(selected.filter(item => item !== group));
+                    setIsOpenGroup(false);
+                }, error => {
+                    show_unexpected_error(error);
+                });
+    };
+
+    const addGroup = group => {
+        return cockpit.spawn(["/usr/sbin/usermod", "-a", "-G", group, name], { superuser: "require", err: "message" })
+                .then(() => {
+                    setSelected([...selected, group]);
+                    setIsOpenGroup(false);
+                }, error => {
+                    show_unexpected_error(error);
+                });
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17996-20221208-201202-d4bd1d46-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)